sdk/src/main/java/io/opentelemetry/sdk/common/export/CompletableResultCode.java

```diff
@@ -0,0 +1,93 @@
+package io.opentelemetry.sdk.common.export;
+
+import java.util.ArrayList;
+import javax.annotation.concurrent.GuardedBy;
+
+/**
+ * The implementation of Export operations are often asynchronous in nature, hence the need to
+ * convey a result at a later time. CompletableResultCode facilitates this.
+ *
+ * <p>This class models JDK 8's CompletableFuture to afford migration should Open Telemetry's SDK
+ * select JDK 8 or greater as a baseline, and also to offer familiarity to developers.
+ */
+public class CompletableResultCode {
+  /** A convenience for declaring success */
+  public static CompletableResultCode ofSuccess() {
+    return SUCCESS;
+  }
+
+  /** A convenience for declaring failure */
+  public static CompletableResultCode ofFailure() {
+    return FAILURE;
+  }
+
+  private static final CompletableResultCode SUCCESS = new CompletableResultCode().success();
+  private static final CompletableResultCode FAILURE = new CompletableResultCode().failure();
+
+  public CompletableResultCode() {}
+
+  @GuardedBy("lock")
+  private Boolean succeeded = null;
+
+  @GuardedBy("lock")
+  private final ArrayList<Runnable> actions = new ArrayList<>();
+
+  private final Object lock = new Object();
+
+  /** The export operation finished successfully. */
+  public CompletableResultCode success() {
```

Comment:
Thanks last nit, these methods looks weird at the call sites since they're not verbs. Something like

`succeed` / `fail`
`complete` / `completeExceptionally`

?

Reply:
`succeed`/`fail` ok? They are both completions, and we're not completing with an exception...

Reply:
Yes, we need unit tests... I'll put a TODO section in the summary and convert this to a draft PR.